src/main/java/com/cs3365/rpgdapi/weapons/WeaponsServiceImpl.java

```diff
@@ -28,4 +28,14 @@ public UUID createWeapon(Weapon requestBody) throws WeaponsException {
 
         return createdWeapon.getId();
     }
+    
+    @Override
+    public WeaponEntity findWeapon(UUID identity) throws WeaponsException
```

Comment:
It's probably not bad to return `WeaponEntity` here instead of weapon, especially because we want to return a response that has the `id` attribute in the body. If this were a project for work, I would request for you to create a new class like `WeaponResponse` that would define what a response body should look like to the end user. I think this is probably okay though. At least for now.

Reply:
Ok. Let me know if I need to change it.